locker.go

```diff
@@ -0,0 +1,52 @@
+package osquery
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type locker struct {
```

Comment:
Could you add a small amount of documentation re: why we need this? `locker is used by the osquery extension client to safely use the thrift socket; we use locker over a mutex to allow for configurable timeouts` or something